File: popconback/src/main/java/com/example/popconback/gifticon/domain/Gifticon.java
package com.example.popconback.gifticon.domain;

import com.example.popconback.files.domain.InputFile;
import com.example.popconback.user.domain.User;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Getter;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@Table(name="gifticon")
public class Gifticon {

    @Id
    @Column(name="barcode_num")
    private String barcodeNum;

    @ManyToOne
    @JoinColumn(name="hash")
    @JsonManagedReference
    private User user;

    @ManyToOne
    @JoinColumn(name="brand_name")
    @JsonManagedReference
    private Brand brand;
    private String productName;
    private Date due;
    private int price;
    @JsonManagedReference
    @OneToMany(mappedBy = "gifticon", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<InputFile> filesList = new ArrayList<>();



    private int state;
    private String memo;


    private int isVoucher;

    @OneToOne(mappedBy = "gifticon", cascade = CascadeType.REMOVE)// 여기가 문제 였음
    private Present present;

}

File: android/POPCON/wear/src/main/java/com/ssafy/popcon/dto/Gifticon.kt
package com.ssafy.popcon.dto

import java.io.Serializable

data class Gifticon(
    var barcodeNum: String,
    val barcode_filepath: String,
    var brand: Brand?,
    var due: String, //example: 2023-01-10 00:00:00.000000
    val hash: Int,
    var price: Int?, //금액권 아니면 -1
    var memo: String,
    val origin_filepath: String,
    var productName: String,
    val product_filepath: String,
    var state : Int
) : Serializable

data class Brand(
    val brandImg: String?,
    var brandName: String
) {
    override fun equals(other: Any?): Boolean {
        return if (other is Brand)
            other.brandName == this.brandName
        else
            false
    }

    override fun hashCode(): Int {
        return brandName.hashCode()
    }
}

data class Badge(
    val content: String,
    val color: String
)

